### ISSUE TITLE
all slices predicted the same

### ISSUE THREAD

**Issue by saksham219** [despoisj/DeepAudioClassification#14]
I have implemented this and trained my model to have a val_acc = 80 % but when I use model.predict() on a new slice it always gives the softmax output with the same class having the same probability. Any reason why this is happening?

**Comment by despoisj** [comment]
Are you sure the validation accuracy is 80% not the training accuracy?
There shouldn't be any difference between the validation and test accuracy
if you manually split your dataset.

Julien
2017-11-06 19:00 GMT+01:00 Saksham Malhotra <notifications@github.com>:

**Comment by saksham219** [comment]
I used the train-test split not the train-test-validation split. The training accuracy came out to be 88 % after 15 epochs and the val_accuracy came out to be 80 percent. I saved the model weights then, loaded the model, took another song, took its slices and fed it into the network and it gave the same class for each slice.

Just another question, which library did you use to create spectrograms? I used librosa

**Comment by despoisj** [comment]
If your validation accuracy is 80% there is definitely something wrong. 
One plausible issue is that your classes might be unbalanced. If you have trained/validated let's say with a vast majority of class A and a minority of class B, the model learned to classify everything as A and still gets a very high accuracy at both training and testing. Take a look at your classes distribution, you might find the culprit!

I used SOX (as seen [here](https://github.com/despoisj/DeepAudioClassification/blob/master/songToData.py))

Good luck!

**Comment by saksham219** [comment]
![me london](https://user-images.githubusercontent.com/28567402/34321935-795ffbd2-e840-11e7-8df1-9aa73c28f458.png)

Hi, I just wanted to know did your spectrograms of the complete songs look like this?

**Comment by despoisj** [comment]
From my memory yes they did, only here you have two lines corresponding to the Left and Right channels of the stereo, and used a RGB color coding. I convert the tracks to mono, and use a Grayscale color code to reduce the input dimensionality.

**Comment by saksham219** [comment]
You have been of great help. Thank you so much. 👍
